git-branchless-revset/src/builtins.rs

```diff
@@ -178,8 +180,57 @@ fn fn_heads(ctx: &mut Context, name: &str, args: &[Expr]) -> EvalResult {
 
 #[instrument]
 fn fn_branches(ctx: &mut Context, name: &str, args: &[Expr]) -> EvalResult {
-    eval0(ctx, name, args)?;
-    Ok(ctx.dag.branch_commits.clone())
+    let pattern = match eval0_or_1_pattern(ctx, name, args)? {
+        Some(pattern) => pattern,
+        None => return Ok(ctx.dag.branch_commits.clone()),
+    };
+
+    let all_branch_commits: HashSet<_> = ctx
+        .dag
+        .commit_set_to_vec(&ctx.dag.branch_commits)
+        .map_err(EvalError::OtherError)?
+        .into_iter()
+        .collect();
```

Comment:
I converted `branch_commits` from `NameSet` to a `HashSet` to get around some waaay over-my-head compiler errors about calling `.contains()` on a `NameSet` and how that's actually a `Pin<Box<dyn Future<...>>>`. In this case, I don't think we need any Eden DAG magic, but I'm open to suggestions if there's a better way to do this.

Reply:
I don't see this in the latest code, but there exists a `Dag::set_contains` method that wraps the `Future` for you that you could use in this case.